src/snowcli/cli/project/util.py

```diff
@@ -18,14 +21,61 @@ def clean_identifier(input):
     return re.sub(r"[^a-z0-9_$]", "", f"{input}".lower())
 
 
+def is_valid_unquoted_identifier(identifier: str) -> bool:
+    """
+    Determines whether the provided identifier is a valid Snowflake unquoted identifier.
+    """
+    return re.fullmatch(UNQUOTED_IDENTIFIER_REGEX, identifier) is not None
+
+
+def is_valid_quoted_identifier(identifier: str) -> bool:
```

Comment:
I think this might return `True` for `""`, can you confirm?

Reply:
Believe it or not, that matches GS behaviour. You can create a database named "", it's a valid ID.

Reply:
Mind blown. Thanks for testing!